Modifications made to the docker client

schaffung opened the pull request:
Following modifications have been made to the code.
1. Changes to the folder structure to accommodate further development ( Addition of  json-c and other functions to the docker client )
2. deb package creation for distribution of the libdocker.
3. Header file to be installed in the /usr/include directory to enable the user to use the functions for further development.
4. Example added with libdocker and option provided for building the example along with libdocker.

schaffung committed:
Created inc director

code changes pending

schaffung committed:
Added an empty example.c in example directory

schaffung committed:
Create c_cpp_properties.json

schaffung committed:
Modified gitignore to ignore .vscode and reoved .vscode.

schaffung committed:
Removed C extern statement from the Header.

schaffung committed:
Modified the project folder structure

Main CMake modified to add .deb package creation options.

schaffung committed:
Update README.md

schaffung committed:
Merge pull request #1 from schaffung/dev

Dev

schaffung committed:
Update README.md

As example directory is being added, removing the example from readme.md

schaffung committed:
Added a compile script for compilation

Created the CMakeLists.txt for the source directory for building the .so for cdockerinterface

schaffung committed:
Merge branch 'dev' of github.com:schaffung/libdocker into dev

schaffung committed:
deb package creation added

Example executable added

compile script added

schaffung committed:
Merge pull request #2 from schaffung/dev

Dev

danielsuo commented:
Thanks for the contribution! In general, I agree with the changes, but a few things:

1. Would change the directory hierarchy back to `src/docker.c`, `inc/docker.h` or put both in `src`
2. We don't need so many levels of `CMakeLists.txt`. A single one where we can develop and release makes more sense
3. Would move the compile script to a `scripts` directory or just remove. `cmake` can handle these operations.

danielsuo commented on README.md:
````diff
@@ -1,4 +1,4 @@
-# ```libdocker```: an SDK for the Docker Engine API in C
+# ```libdocker```: an SDK for the Docker Engine API in C ( Hence removed the compatibility with cpp )
````
What's the reason for removing compatibility with `cpp`? Would prefer to keep in unless there's a good reason.

schaffung replied:
The Docker website cites libdocker as an unofficial C SDK. For C++, they have quoted some other library. So my reason for the removal of the compatibility was to strictly stick with the mentioned purpose, i.e. C SDK.

danielsuo commented on README.md:
````diff
@@ -10,60 +10,17 @@ A minimalist SDK for accessing the [Docker Engine API](https://docs.docker.com/e
+ they can manually edit the base CMakeLists. In future, the compile script will accomodate the option.)
+- If the build is successful, then a deb package will be created in the build directory. Use the command ```sudo dpkg -i <deb package name>```
+to install the package.
+- This will add libcdockerinterface to the /usr/lib directory and the example executable cdockerinterfaceexample in /usr/bin
````
These names are mouthfuls; I understand where you're going with them, but can we keep the name or is there conceptual/actual namespace collision?

schaffung replied:
No there is no collision.

danielsuo commented on source_code/CDOCKER/CMakeLists.txt:
```diff
@@ -0,0 +1,19 @@
+    ${CMAKE_CURRENT_LIST_DIR}/src/docker.c
+)
+
+add_library( cdockerinterface SHARED ${SRC_FILES})
```
See comment above re naming. I would prefer `docker` or, if it's too confusing, `cdocker`

schaffung replied:
Ok. I'll modify the naming.

ghost commented:
Sure. The changes sound good. I'll make these changes and then raise a pull request..

Thanks for the quick response....

danielsuo commented on source_code/CDOCKER/inc/docker.h:
```diff
@@ -11,10 +11,6 @@
 #include <string.h>
 #include <curl/curl.h>
 
-#ifdef __cplusplus
```
See comment above re removing C++ support.

danielsuo commented:
Wait, you don't need to close and recreate the pull request; you can just keep committing to this branch.

schaffung committed:
Reduced the CMake to one level.

Moved the compile to a scripts directory

simplified the naming of the .so and the executable

folder structure of the project simplified

schaffung committed:
Merge pull request #3 from schaffung/dev

Changes made according to reflect the requested corrections from Upstream

schaffung committed:
Update README.md

Changed the name of the .so and executable.

ghost commented:
Hi...

I have made the following changes,
1. Reducing the ```cmake``` file to one.
2. Changing the .so and executable names to cdocker and cdocker_example.
3. Moved the docker.c and docker.h to inc/docker.h and src/docker.c respectivley.

Please let me know if the cpp compatibility is to be added. I'll make changes accordingly.

danielsuo commented:
Yeah, I would leave it in because it's generally something c libraries have.

On Wed, Mar 25, 2020 at 10:44 AM Srijan <notifications@github.com> wrote:

> Hi...
>
> I have made the following changes,
>
>    1. Reducing the cmake file to one.
>    2. Changing the .so and executable names to cdocker and
>    cdocker_example.
>    3. Moved the docker.c and docker.h to inc/docker.h and src/docker.c
>    respectivley.
>
> Please let me know if the cpp compatibility is to be added. I'll make
> changes accordingly.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/danielsuo/libdocker/pull/9#issuecomment-603880068>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAURLVVHVI4NCGMIY2FYOXDRJIKF5ANCNFSM4LTOZM6A>
> .
>

schaffung committed:
cpp compatibility

Added the cpp compatibility which was removed earlier

schaffung committed:
Merge pull request #4 from schaffung/dev

cpp compatibility

ghost commented:
I have added back the cpp compatibility.

schaffung committed:
Update README.md

danielsuo commented on CMakeLists.txt:
```diff
@@ -1,15 +1,82 @@
-# Set target name and source files
-file (GLOB DOCKER_SOURCE_FILES RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} src/**)
+# Project name.
+project(cdocker)
```
A bit of a stickler -- given that the repository name and the link on the Docker website is to libdocker, can we just keep everything to docker instead of cdocker? If you feel strongly, we should probably rename repository and talk to Docker about changing their listing to keep everything consistent.

schaffung replied:
I think docker could also work but the only hiccup I though could arise is the presence of docker command which is also ```docker```. But anyway, .so won't have any conflict with that, so I think changing ```cdocker``` to ```docker``` sounds sane.

schaffung committed:
Modified naming from cdocker to docker

1. In cmake project name

2. In .so name

3. .deb package name modified to docker_c_sdk

4. executable name changed to docker_example

schaffung committed:
Merge pull request #5 from schaffung/dev

Modified naming from cdocker to docker

ghost commented:
Modified the naming. But please note that for the example executable, I've kept the name as docker_example ( I think a better name can be given ) and the deb package base name is docker_c_sdk.

You can suggest if these names have to be modified...

schaffung committed:
Update README.md

schaffung committed:
Update README.md
